datalad/distribution/publish.py

```diff
@@ -343,11 +348,25 @@ def __call__(
                         'Cannot determine target sibling for %s' % (ds,))
             elif to not in ds.repo.get_remotes():
                 # unknown given remote
+                # TODO: not quite "failing" really so interfers with "inherit_settings"
                 if skip_failing:
                     lgr.warning(
                         "Unknown target sibling '%s', skipping %s",
                         to, ds)
                     ds_remote_info[ds_path] = None
+                elif inherit_settings:
+                    superds = ds.get_superdataset()
+                    if not superds:
+                        raise RuntimeError(
+                            "%s has no super-dataset to inherit settings for the remote %s"
+                            % (ds, to)
+                        )
+                    # XXX due to difference between create-sibling and create-sibling-github
+                    # would not be as transparent to inherit for -github
+                    lgr.info("Will try to create a sibling inheriting settings from %s", superds)
+                    # XXX explicit None as sshurl for now
+                    ds.create_sibling(None, name=to, inherit_settings=True)
```

Comment:
May be we just need more detailed values for `inherit` to distinguish whether we should try to create or just add that sibling.